zstd/encoder_options.go

```diff
@@ -316,3 +322,14 @@ func WithEncoderDict(dict []byte) EOption {
 		return nil
 	}
 }
+
+// WithEncoderDictRaw registers a dictionary that may be used by the encoder.
+//
+// The slice content may contain arbitrary data. It will be used as an initial
+// history.
+func WithEncoderDictRaw(id uint32, content []byte) EOption {
+	return func(o *encoderOptions) error {
+		o.dict = &dict{id: id, content: content}
```

Comment:
offsets must be initialized to `[3]uint32{1, 4, 8}`.

Reply:
Done. Also put in a check for dictionaries that are too large for the reference implementation.